src/api.rs

```diff
@@ -154,7 +154,8 @@ impl MiddleWare for EndpointMiddleware {
         let mut url_c = url.clone();
         let mut segs: Vec<&str> = url.path_segments().unwrap().collect();
         segs.insert(0, self.version.as_str());
-        url_c.path_segments_mut().unwrap().clear().extend(segs);
+        url_c.set_path("");
+        url_c = url_c.join(&segs.join("/")).unwrap();
```

Comment:
I'm not exactly sure, if this is the smoothest way but seems to work well, as `Url::join` won't encode again.